core/src/banking_stage.rs

```diff
@@ -376,13 +379,221 @@ pub enum ForwardOption {
     ForwardTransaction,
 }
 
-#[derive(Debug)]
+#[derive(Debug, Default)]
 pub struct FilterForwardingResults {
     total_forwardable_packets: usize,
     total_tracer_packets_in_buffer: usize,
     total_forwardable_tracer_packets: usize,
 }
 
+#[derive(Debug, PartialEq, Eq, Copy, Clone)]
+pub enum ThreadType {
+    Voting(VoteSource),
+    Transactions,
+}
+
+#[derive(Debug)]
+pub enum UnprocessedTransactionStorage {
+    VoteStorage(Arc<LatestUnprocessedVotes>, VoteSource),
+    TransactionStorage(UnprocessedPacketBatches, ThreadType),
+}
+
+impl UnprocessedTransactionStorage {
+    pub fn is_empty(&self) -> bool {
+        match self {
+            Self::VoteStorage(v, _) => v.is_empty(),
+            Self::TransactionStorage(t, _) => t.is_empty(),
+        }
+    }
+
+    pub fn len(&self) -> usize {
+        match self {
+            Self::VoteStorage(v, _) => v.len(),
+            Self::TransactionStorage(t, _) => t.len(),
+        }
+    }
+
+    pub fn capacity(&self) -> usize {
+        match self {
+            Self::VoteStorage(_, _) => usize::MAX,
```

Comment:
Just curious, would there be need to limit number of queued votes?

Reply:
technically it is limited by the # of voters